backend/app/api/triggers.py

```
@@ -0,0 +1,185 @@
+        
+        db.add(db_trigger)
+        db.commit()
+        db.refresh(db_trigger)
+        
+        return TriggerResponse.from_orm(db_trigger)
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error creating trigger: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to create trigger: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)